components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMConfigProcessorTest.java

```diff
@@ -0,0 +1,121 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.scim2.common.utils;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.charon3.core.exceptions.CharonException;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class SCIMConfigProcessorTest {
+    private SCIMConfigProcessor scimConfigProcessor;
+    private AuthenticationSchema authenticationSchema;
+
+    @DataProvider(name = "facebook-authenticator-all") public static Object[][] propertyProvider() {
+        return new Object[][] { { "Name", "SCIM2" }, { "Address", null } };
+    }
+
+    @DataProvider() public static Object[][] filePathProvider() {
+        String errorFileNamePath = Paths
+                .get(System.getProperty("user.dir"), "src", "test", "resources", "charon-config-tst.xml").toString();
+
+        String noFilepath = Paths.get(System.getProperty("user.dir"), "src", "resources", "charon-config-test.xml")
+                .toString();
+
+        String errorFilePath = Paths
+                .get(System.getProperty("user.dir"), "srcTest", "test", "resources", "charon-config-test.xml")
+                .toString();
+        return new Object[][] { { noFilepath }, { errorFilePath }, { errorFileNamePath } };
+    }
+
+    @BeforeMethod public void setUp() throws Exception {
+        scimConfigProcessor = new SCIMConfigProcessor();
+
+        scimConfigProcessor.properties.put("Name", "SCIM2");
+        scimConfigProcessor.properties.put("Age", "24");
+
+        authenticationSchema = new AuthenticationSchema();
+        authenticationSchema.setName("SCIM");
+        authenticationSchema.setDescription("SCIM2");
+        authenticationSchema.setSpecUri("https://localhost:9443/scim2");
+        authenticationSchema.setDocumentationUri("https://localhost:9443/scim2/docs");
+        authenticationSchema.setType("Authentication");
+        authenticationSchema.setPrimary("true");
+        scimConfigProcessor.authenticationSchemas = new ArrayList<>();
+        scimConfigProcessor.authenticationSchemas.add(authenticationSchema);
+    }
+
+    @AfterMethod public void tearDown() throws Exception {
+    }
+
+    @Test public void testGetProperties() throws Exception {
+        Map<String, String> expected = new HashMap<String, String>();
+        expected.put("Name", "SCIM2");
+        expected.put("Age", "24");
+
+        Map<String, String> map = scimConfigProcessor.getProperties();
+
+        assertEquals(map.get("Name"), expected.get("Name"));
+        assertEquals(map.get("Age"), expected.get("Age"));
+    }
+
+    @Test(dataProvider = "facebook-authenticator-all") public void testGetProperty(String property,
```

Comment:
I think it's better to have the @Test annotation and the method declaration in two different lines.

Reply:
When I format the code from the wso2 code styles, the code becomes like that.